### PR TITLE
chore: add engines to package.json

### PR DESCRIPTION
**Pull request by ju-Skinner**
# Description
This assists developers with knowing which node versions are compatible with the Design System. 

Please delete options that are not relevant. 
If your type of change is not present, add that option.
- [X] Chore (other changes that don't affect source or test files)

# How Has This Been Tested?
N/A

**Test Configuration**:

- Sage versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR
- [X] No options apply

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,10 +1,12 @@
 {
-  "name": "sage",
   "private": true,
   "workspaces": [
     "apps/*",
     "libs/*"
   ],
+  "engines": {
+    "node": ">= 16"
```
Comment (voodooGQ) [inline package.json:8]: Any reason we're starting with 16 instead of 18? 16 stopped receiving security support on 10-18-2022

Comment (ju-Skinner) [reply package.json:8]: Yes, you are right. Does that mean that people should be forced to use 18 to develop within our ecosystem? If I recall, we are saying that we know that our code works with this minimum version of node. End of Life for 16 is later this year, you can find out more [here](https://nodejs.dev/en/about/releases/)

**Review by rwc9u — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (voodooGQ) [reply package.json:8]: You're correct, but with that stance we have to have some sort of testing scenarios in place to verify 16 and 18 both work (probably from CI). If that's in the plans then I think this is totally reasonable, if not I'd say that we should probably put it on the docket.

Comment (ju-Skinner) [reply package.json:8]: See the latest [commit](https://github.com/Kajabi/sage/pull/50/commits/a94f8d2ec4e3ddead98d190bdf831b054eabc509)

**Review by voodooGQ — APPROVED** [review]

**Pull request merged**